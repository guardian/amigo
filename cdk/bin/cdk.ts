import { App } from "aws-cdk-lib";
import type { AmigoProps } from "../lib/amigo";
import { AmigoStack } from "../lib/amigo";

const app = new App();

const stageAgnosticProps = {
  stack: "deploy",
  migratedFromCloudFormation: true,
};

const amigoCodeProps: AmigoProps = {
  ...stageAgnosticProps,
  stage: "CODE",
  domainName: "amigo.code.dev-gutools.co.uk",
};

new AmigoStack(app, "AMIgo-CODE", amigoCodeProps);

export const amigoProdProps: AmigoProps = {
  ...stageAgnosticProps,
  stage: "PROD",
  domainName: "amigo.gutools.co.uk",
};

new AmigoStack(app, "AMIgo-PROD", amigoProdProps);
