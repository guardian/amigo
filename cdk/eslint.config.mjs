import guardian from '@guardian/eslint-config';

export default [
	...guardian.configs.recommended,
	...guardian.configs.jest
];