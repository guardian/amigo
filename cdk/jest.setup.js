/* eslint-disable no-undef -- jest globals are missing due to guardian.recommended.jest eslint rules not covering jest.setup.js */
jest.mock("@guardian/cdk/lib/constants/tracking-tag");
