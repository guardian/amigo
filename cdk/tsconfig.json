{
  "extends": "@guardian/tsconfig/tsconfig.json",
  "compilerOptions": {
    "module": "CommonJS"
  },
  "include": ["lib/**/*", "bin/**/*"],
  "exclude": [
    "node_modules",
    "cdk.out",
    "lib/**/__snapshots__/**"
  ]
}
