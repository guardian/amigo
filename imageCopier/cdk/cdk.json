{
  "app": "npx ts-node lib/app.ts",
  "context": {
    "aws-cdk:enableDiffNoFail": "true",
    "@aws-cdk/core:stackRelativeExports": "true"
  },
  "//": "~~ Generated by projen. To modify, edit .projenrc.js and run \"npx projen\"."
}
