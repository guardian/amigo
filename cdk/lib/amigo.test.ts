import { App } from "aws-cdk-lib";
import { Template } from "aws-cdk-lib/assertions";
import { amigoProdProps } from "../bin/cdk";
import { AmigoStack } from "./amigo";

describe("The Amigo stack", () => {
  it("matches the snapshot", () => {
    const app = new App();
    const stack = new AmigoStack(app, "AMIgo", amigoProdProps);
    expect(Template.fromStack(stack).toJSON()).toMatchSnapshot();
  });
});
