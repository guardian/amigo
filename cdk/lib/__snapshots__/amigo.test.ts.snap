// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Amigo stack matches the snapshot 1`] = `
Object {
  "Mappings": Object {
    "amigo": Object {
      "CODE": Object {
        "DataBucketName": "amigo-data-code",
        "domainName": "amigo.code.dev-gutools.co.uk",
        "maxInstances": 2,
        "minInstances": 1,
      },
      "PROD": Object {
        "DataBucketName": "amigo-data-prod",
        "domainName": "amigo.gutools.co.uk",
        "maxInstances": 2,
        "minInstances": 1,
      },
    },
  },
  "Outputs": Object {
    "LoadBalancerAmigoDnsName": Object {
      "Description": "DNS entry for LoadBalancerAmigo",
      "Value": Object {
        "Fn::GetAtt": Array [
          "LoadBalancerAmigoC3017FAF",
          "DNSName",
        ],
      },
    },
  },
  "Parameters": Object {
    "AMIAmigo": Object {
      "Description": "Amazon Machine Image ID for the app amigo. Use this in conjunction with AMIgo to keep AMIs up to date.",
      "Type": "AWS::EC2::Image::Id",
    },
    "AnghammaradSnsArn": Object {
      "Default": "/account/services/anghammarad.topic.arn",
      "Description": "SSM parameter containing the ARN of the Anghammarad SNS topic",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
    "DistributionBucketName": Object {
      "Default": "/account/services/artifact.bucket",
      "Description": "SSM parameter containing the S3 bucket name holding distribution artifacts",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
    "LoggingStreamName": Object {
      "Default": "/account/services/logging.stream.name",
      "Description": "SSM parameter containing the Name (not ARN) on the kinesis stream",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
    "PackerInstanceProfile": Object {
      "Description": "Instance profile given to instances created by Packer. Find this in the PackerUser-PackerRole in IAM",
      "Type": "String",
    },
    "Stage": Object {
      "AllowedValues": Array [
        "CODE",
        "PROD",
      ],
      "Default": "CODE",
      "Description": "Stage name",
      "Type": "String",
    },
    "VpcId": Object {
      "Default": "/account/vpc/primary/id",
      "Description": "Virtual Private Cloud to run EC2 instances within",
      "Type": "AWS::SSM::Parameter::Value<AWS::EC2::VPC::Id>",
    },
    "amigoPrivateSubnets": Object {
      "Default": "/account/vpc/primary/subnets/private",
      "Description": "A list of private subnets",
      "Type": "AWS::SSM::Parameter::Value<List<AWS::EC2::Subnet::Id>>",
    },
    "amigoPublicSubnets": Object {
      "Default": "/account/vpc/primary/subnets/public",
      "Description": "A list of public subnets",
      "Type": "AWS::SSM::Parameter::Value<List<AWS::EC2::Subnet::Id>>",
    },
  },
  "Resources": Object {
    "AmigoCname": Object {
      "Properties": Object {
        "Name": Object {
          "Fn::FindInMap": Array [
            "amigo",
            Object {
              "Ref": "Stage",
            },
            "domainName",
          ],
        },
        "RecordType": "CNAME",
        "ResourceRecords": Array [
          Object {
            "Fn::GetAtt": Array [
              "LoadBalancerAmigoC3017FAF",
              "DNSName",
            ],
          },
        ],
        "Stage": Object {
          "Ref": "Stage",
        },
        "TTL": 3600,
      },
      "Type": "Guardian::DNS::RecordSet",
    },
    "AmigoDataBucket": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "BucketName": Object {
          "Fn::FindInMap": Array [
            "amigo",
            Object {
              "Ref": "Stage",
            },
            "DataBucketName",
          ],
        },
        "PublicAccessBlockConfiguration": Object {
          "BlockPublicAcls": true,
          "BlockPublicPolicy": true,
          "IgnorePublicAcls": true,
          "RestrictPublicBuckets": true,
        },
        "Tags": Array [
          Object {
            "Key": "App",
            "Value": "amigo",
          },
          Object {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          Object {
            "Key": "gu:repo",
            "Value": "guardian/amigo",
          },
          Object {
            "Key": "Stack",
            "Value": "deploy",
          },
          Object {
            "Key": "Stage",
            "Value": Object {
              "Ref": "Stage",
            },
          },
        ],
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Retain",
    },
    "AppPolicyF941AEC5": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "s3:GetObject",
              "Effect": "Allow",
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    Object {
                      "Fn::GetAtt": Array [
                        "AmigoDataBucket",
                        "Arn",
                      ],
                    },
                    "/*",
                  ],
                ],
              },
            },
            Object {
              "Action": "dynamodb:ListTables",
              "Effect": "Allow",
              "Resource": "*",
            },
            Object {
              "Action": "dynamodb:*",
              "Effect": "Allow",
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    "arn:aws:dynamodb:*:*:table/amigo-",
                    Object {
                      "Ref": "Stage",
                    },
                    "-*",
                  ],
                ],
              },
            },
            Object {
              "Action": "sns:ListTopics",
              "Effect": "Allow",
              "Resource": "*",
            },
            Object {
              "Action": "sns:*",
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:aws:sns:*:*:amigo-",
                      Object {
                        "Ref": "Stage",
                      },
                      "-notify",
                    ],
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:aws:sns:*:*:amigo-",
                      Object {
                        "Ref": "Stage",
                      },
                      "-housekeeping-notify",
                    ],
                  ],
                },
              ],
            },
            Object {
              "Action": Array [
                "s3:GetBucketPolicy",
                "s3:PutBucketPolicy",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    "arn:aws:s3::*:",
                    Object {
                      "Ref": "DistributionBucketName",
                    },
                  ],
                ],
              },
            },
            Object {
              "Action": "iam:GetInstanceProfile",
              "Effect": "Allow",
              "Resource": Object {
                "Ref": "PackerInstanceProfile",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "app-policy",
        "Roles": Array [
          Object {
            "Ref": "InstanceRoleAmigo75944747",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "AutoScalingGroupAmigoASG5BE72894": Object {
      "Properties": Object {
        "HealthCheckGracePeriod": 120,
        "HealthCheckType": "ELB",
        "LaunchConfigurationName": Object {
          "Ref": "AutoScalingGroupAmigoLaunchConfig93624D33",
        },
        "MaxSize": Object {
          "Fn::FindInMap": Array [
            "amigo",
            Object {
              "Ref": "Stage",
            },
            "maxInstances",
          ],
        },
        "MinSize": Object {
          "Fn::FindInMap": Array [
            "amigo",
            Object {
              "Ref": "Stage",
            },
            "minInstances",
          ],
        },
        "Tags": Array [
          Object {
            "Key": "App",
            "PropagateAtLaunch": true,
            "Value": "amigo",
          },
          Object {
            "Key": "gu:cdk:pattern-name",
            "PropagateAtLaunch": true,
            "Value": "GuPlayApp",
          },
          Object {
            "Key": "gu:cdk:version",
            "PropagateAtLaunch": true,
            "Value": "TEST",
          },
          Object {
            "Key": "gu:repo",
            "PropagateAtLaunch": true,
            "Value": "guardian/amigo",
          },
          Object {
            "Key": "Name",
            "PropagateAtLaunch": true,
            "Value": "AMIgo/AutoScalingGroupAmigo",
          },
          Object {
            "Key": "Stack",
            "PropagateAtLaunch": true,
            "Value": "deploy",
          },
          Object {
            "Key": "Stage",
            "PropagateAtLaunch": true,
            "Value": Object {
              "Ref": "Stage",
            },
          },
        ],
        "TargetGroupARNs": Array [
          Object {
            "Ref": "TargetGroupAmigoB9501F07",
          },
        ],
        "VPCZoneIdentifier": Object {
          "Ref": "amigoPrivateSubnets",
        },
      },
      "Type": "AWS::AutoScaling::AutoScalingGroup",
    },
    "AutoScalingGroupAmigoInstanceProfileED0FC3E0": Object {
      "Properties": Object {
        "Roles": Array [
          Object {
            "Ref": "InstanceRoleAmigo75944747",
          },
        ],
      },
      "Type": "AWS::IAM::InstanceProfile",
    },
    "AutoScalingGroupAmigoLaunchConfig93624D33": Object {
      "DependsOn": Array [
        "InstanceRoleAmigo75944747",
      ],
      "Properties": Object {
        "IamInstanceProfile": Object {
          "Ref": "AutoScalingGroupAmigoInstanceProfileED0FC3E0",
        },
        "ImageId": Object {
          "Ref": "AMIAmigo",
        },
        "InstanceType": "t4g.small",
        "SecurityGroups": Array [
          Object {
            "Fn::GetAtt": Array [
              "GuHttpsEgressSecurityGroupAmigo28861996",
              "GroupId",
            ],
          },
          Object {
            "Fn::GetAtt": Array [
              "WazuhSecurityGroup",
              "GroupId",
            ],
          },
        ],
        "UserData": Object {
          "Fn::Base64": Object {
            "Fn::Join": Array [
              "",
              Array [
                "#!/bin/bash -ev
wget -P /tmp https://releases.hashicorp.com/packer/1.6.6/packer_1.6.6_linux_arm64.zip
mkdir /opt/packer
unzip -d /opt/packer /tmp/packer_*_linux_arm64.zip
echo 'export PATH=\${!PATH}:/opt/packer' > /etc/profile.d/packer.sh
wget -P /tmp https://s3.amazonaws.com/session-manager-downloads/plugin/latest/ubuntu_arm64/session-manager-plugin.deb
dpkg -i /tmp/session-manager-plugin.deb
aws --region eu-west-1 s3 cp s3://",
                Object {
                  "Ref": "DistributionBucketName",
                },
                "/deploy/",
                Object {
                  "Ref": "Stage",
                },
                "/amigo/amigo_1.0-latest_all.deb /tmp/amigo.deb
dpkg -i /tmp/amigo.deb",
              ],
            ],
          },
        },
      },
      "Type": "AWS::AutoScaling::LaunchConfiguration",
    },
    "CertificateAmigoE1AF5E0C": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "DomainName": Object {
          "Fn::FindInMap": Array [
            "amigo",
            Object {
              "Ref": "Stage",
            },
            "domainName",
          ],
        },
        "Tags": Array [
          Object {
            "Key": "App",
            "Value": "amigo",
          },
          Object {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          Object {
            "Key": "gu:repo",
            "Value": "guardian/amigo",
          },
          Object {
            "Key": "Stack",
            "Value": "deploy",
          },
          Object {
            "Key": "Stage",
            "Value": Object {
              "Ref": "Stage",
            },
          },
        ],
        "ValidationMethod": "DNS",
      },
      "Type": "AWS::CertificateManager::Certificate",
      "UpdateReplacePolicy": "Retain",
    },
    "DescribeEC2PolicyFF5F9295": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "autoscaling:DescribeAutoScalingInstances",
                "autoscaling:DescribeAutoScalingGroups",
                "ec2:DescribeTags",
                "ec2:DescribeInstances",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "describe-ec2-policy",
        "Roles": Array [
          Object {
            "Ref": "InstanceRoleAmigo75944747",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "GetDistributablePolicyAmigoB25A5D2B": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "s3:GetObject",
              "Effect": "Allow",
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    "arn:aws:s3:::",
                    Object {
                      "Ref": "DistributionBucketName",
                    },
                    "/deploy/",
                    Object {
                      "Ref": "Stage",
                    },
                    "/amigo/*",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "GetDistributablePolicyAmigoB25A5D2B",
        "Roles": Array [
          Object {
            "Ref": "InstanceRoleAmigo75944747",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "GuAnghammaradSenderPolicy674A3874": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sns:Publish",
              "Effect": "Allow",
              "Resource": Object {
                "Ref": "AnghammaradSnsArn",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "GuAnghammaradSenderPolicy674A3874",
        "Roles": Array [
          Object {
            "Ref": "InstanceRoleAmigo75944747",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "GuHttpsEgressSecurityGroupAmigo28861996": Object {
      "Properties": Object {
        "GroupDescription": "Allow all outbound HTTPS traffic",
        "SecurityGroupEgress": Array [
          Object {
            "CidrIp": "0.0.0.0/0",
            "Description": "Allow all outbound HTTPS traffic",
            "FromPort": 443,
            "IpProtocol": "tcp",
            "ToPort": 443,
          },
        ],
        "Tags": Array [
          Object {
            "Key": "App",
            "Value": "amigo",
          },
          Object {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          Object {
            "Key": "gu:repo",
            "Value": "guardian/amigo",
          },
          Object {
            "Key": "Stack",
            "Value": "deploy",
          },
          Object {
            "Key": "Stage",
            "Value": Object {
              "Ref": "Stage",
            },
          },
        ],
        "VpcId": Object {
          "Ref": "VpcId",
        },
      },
      "Type": "AWS::EC2::SecurityGroup",
    },
    "GuHttpsEgressSecurityGroupAmigofromAMIgoLoadBalancerAmigoSecurityGroup730D090E90006200512B": Object {
      "Properties": Object {
        "Description": "Load balancer to target",
        "FromPort": 9000,
        "GroupId": Object {
          "Fn::GetAtt": Array [
            "GuHttpsEgressSecurityGroupAmigo28861996",
            "GroupId",
          ],
        },
        "IpProtocol": "tcp",
        "SourceSecurityGroupId": Object {
          "Fn::GetAtt": Array [
            "LoadBalancerAmigoSecurityGroup304D707B",
            "GroupId",
          ],
        },
        "ToPort": 9000,
      },
      "Type": "AWS::EC2::SecurityGroupIngress",
    },
    "GuHttpsEgressSecurityGroupAmigofromAMIgoRestrictedIngressSecurityGroupAmigoE4F60B0990004CAF7118": Object {
      "Properties": Object {
        "Description": "Load balancer to target",
        "FromPort": 9000,
        "GroupId": Object {
          "Fn::GetAtt": Array [
            "GuHttpsEgressSecurityGroupAmigo28861996",
            "GroupId",
          ],
        },
        "IpProtocol": "tcp",
        "SourceSecurityGroupId": Object {
          "Fn::GetAtt": Array [
            "RestrictedIngressSecurityGroupAmigoF34D371D",
            "GroupId",
          ],
        },
        "ToPort": 9000,
      },
      "Type": "AWS::EC2::SecurityGroupIngress",
    },
    "GuLogShippingPolicy981BFE5A": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "kinesis:Describe*",
                "kinesis:Put*",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    "arn:aws:kinesis:",
                    Object {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    Object {
                      "Ref": "AWS::AccountId",
                    },
                    ":stream/",
                    Object {
                      "Ref": "LoggingStreamName",
                    },
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "GuLogShippingPolicy981BFE5A",
        "Roles": Array [
          Object {
            "Ref": "InstanceRoleAmigo75944747",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "InstanceRoleAmigo75944747": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "ec2.",
                      Object {
                        "Ref": "AWS::URLSuffix",
                      },
                    ],
                  ],
                },
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "Path": "/",
        "Tags": Array [
          Object {
            "Key": "App",
            "Value": "amigo",
          },
          Object {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          Object {
            "Key": "gu:repo",
            "Value": "guardian/amigo",
          },
          Object {
            "Key": "Stack",
            "Value": "deploy",
          },
          Object {
            "Key": "Stage",
            "Value": Object {
              "Ref": "Stage",
            },
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "ListenerAmigo0180B42C": Object {
      "Properties": Object {
        "Certificates": Array [
          Object {
            "CertificateArn": Object {
              "Ref": "CertificateAmigoE1AF5E0C",
            },
          },
        ],
        "DefaultActions": Array [
          Object {
            "TargetGroupArn": Object {
              "Ref": "TargetGroupAmigoB9501F07",
            },
            "Type": "forward",
          },
        ],
        "LoadBalancerArn": Object {
          "Ref": "LoadBalancerAmigoC3017FAF",
        },
        "Port": 443,
        "Protocol": "HTTPS",
      },
      "Type": "AWS::ElasticLoadBalancingV2::Listener",
    },
    "LoadBalancerAmigoC3017FAF": Object {
      "Properties": Object {
        "LoadBalancerAttributes": Array [
          Object {
            "Key": "deletion_protection.enabled",
            "Value": "true",
          },
        ],
        "Scheme": "internet-facing",
        "SecurityGroups": Array [
          Object {
            "Fn::GetAtt": Array [
              "LoadBalancerAmigoSecurityGroup304D707B",
              "GroupId",
            ],
          },
          Object {
            "Fn::GetAtt": Array [
              "RestrictedIngressSecurityGroupAmigoF34D371D",
              "GroupId",
            ],
          },
        ],
        "Subnets": Object {
          "Ref": "amigoPublicSubnets",
        },
        "Tags": Array [
          Object {
            "Key": "App",
            "Value": "amigo",
          },
          Object {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          Object {
            "Key": "gu:repo",
            "Value": "guardian/amigo",
          },
          Object {
            "Key": "Stack",
            "Value": "deploy",
          },
          Object {
            "Key": "Stage",
            "Value": Object {
              "Ref": "Stage",
            },
          },
        ],
        "Type": "application",
      },
      "Type": "AWS::ElasticLoadBalancingV2::LoadBalancer",
    },
    "LoadBalancerAmigoSecurityGroup304D707B": Object {
      "Properties": Object {
        "GroupDescription": "Automatically created Security Group for ELB AMIgoLoadBalancerAmigo85BA3984",
        "Tags": Array [
          Object {
            "Key": "App",
            "Value": "amigo",
          },
          Object {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          Object {
            "Key": "gu:repo",
            "Value": "guardian/amigo",
          },
          Object {
            "Key": "Stack",
            "Value": "deploy",
          },
          Object {
            "Key": "Stage",
            "Value": Object {
              "Ref": "Stage",
            },
          },
        ],
        "VpcId": Object {
          "Ref": "VpcId",
        },
      },
      "Type": "AWS::EC2::SecurityGroup",
    },
    "LoadBalancerAmigoSecurityGrouptoAMIgoGuHttpsEgressSecurityGroupAmigo34A5A8389000E5CCAB10": Object {
      "Properties": Object {
        "Description": "Load balancer to target",
        "DestinationSecurityGroupId": Object {
          "Fn::GetAtt": Array [
            "GuHttpsEgressSecurityGroupAmigo28861996",
            "GroupId",
          ],
        },
        "FromPort": 9000,
        "GroupId": Object {
          "Fn::GetAtt": Array [
            "LoadBalancerAmigoSecurityGroup304D707B",
            "GroupId",
          ],
        },
        "IpProtocol": "tcp",
        "ToPort": 9000,
      },
      "Type": "AWS::EC2::SecurityGroupEgress",
    },
    "PackerPolicyE6040EA8": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "ec2:AttachVolume",
                "ec2:AuthorizeSecurityGroupIngress",
                "ec2:CopyImage",
                "ec2:CreateImage",
                "ec2:CreateKeypair",
                "ec2:CreateSecurityGroup",
                "ec2:CreateSnapshot",
                "ec2:CreateTags",
                "ec2:CreateVolume",
                "ec2:DeleteKeypair",
                "ec2:DeleteSecurityGroup",
                "ec2:DeleteSnapshot",
                "ec2:DeleteVolume",
                "ec2:DeregisterImage",
                "ec2:DescribeImageAttribute",
                "ec2:DescribeImages",
                "ec2:DescribeInstances",
                "ec2:DescribeRegions",
                "ec2:DescribeSecurityGroups",
                "ec2:DescribeSnapshots",
                "ec2:DescribeSubnets",
                "ec2:DescribeTags",
                "ec2:DescribeVolumes",
                "ec2:DetachVolume",
                "ec2:GetPasswordData",
                "ec2:ModifyImageAttribute",
                "ec2:ModifyInstanceAttribute",
                "ec2:ModifySnapshotAttribute",
                "ec2:RegisterImage",
                "ec2:RunInstances",
                "ec2:StopInstances",
                "ec2:TerminateInstances",
                "iam:PassRole",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "packer-required-permissions",
        "Roles": Array [
          Object {
            "Ref": "InstanceRoleAmigo75944747",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "PackerSecurityGroup": Object {
      "Properties": Object {
        "GroupDescription": "Security group for instances created by Packer",
        "GroupName": Object {
          "Fn::Join": Array [
            "",
            Array [
              "amigo-packer-",
              Object {
                "Ref": "Stage",
              },
            ],
          ],
        },
        "SecurityGroupEgress": Array [
          Object {
            "CidrIp": "0.0.0.0/0",
            "Description": "Allow all outbound TCP",
            "FromPort": 0,
            "IpProtocol": "tcp",
            "ToPort": 65535,
          },
        ],
        "Tags": Array [
          Object {
            "Key": "App",
            "Value": "amigo",
          },
          Object {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          Object {
            "Key": "gu:repo",
            "Value": "guardian/amigo",
          },
          Object {
            "Key": "Stack",
            "Value": "deploy",
          },
          Object {
            "Key": "Stage",
            "Value": Object {
              "Ref": "Stage",
            },
          },
        ],
        "VpcId": Object {
          "Ref": "VpcId",
        },
      },
      "Type": "AWS::EC2::SecurityGroup",
    },
    "ParameterStoreReadAmigoA9D847D3": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "ssm:GetParametersByPath",
              "Effect": "Allow",
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    "arn:aws:ssm:",
                    Object {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    Object {
                      "Ref": "AWS::AccountId",
                    },
                    ":parameter/",
                    Object {
                      "Ref": "Stage",
                    },
                    "/deploy/amigo",
                  ],
                ],
              },
            },
            Object {
              "Action": Array [
                "ssm:GetParameters",
                "ssm:GetParameter",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    "arn:aws:ssm:",
                    Object {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    Object {
                      "Ref": "AWS::AccountId",
                    },
                    ":parameter/",
                    Object {
                      "Ref": "Stage",
                    },
                    "/deploy/amigo/*",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "parameter-store-read-policy",
        "Roles": Array [
          Object {
            "Ref": "InstanceRoleAmigo75944747",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "RestrictedIngressSecurityGroupAmigoF34D371D": Object {
      "Properties": Object {
        "GroupDescription": "Allow restricted ingress from CIDR ranges",
        "SecurityGroupIngress": Array [
          Object {
            "CidrIp": "77.91.248.0/21",
            "Description": "Allow access on port 443 from 77.91.248.0/21",
            "FromPort": 443,
            "IpProtocol": "tcp",
            "ToPort": 443,
          },
        ],
        "Tags": Array [
          Object {
            "Key": "App",
            "Value": "amigo",
          },
          Object {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          Object {
            "Key": "gu:repo",
            "Value": "guardian/amigo",
          },
          Object {
            "Key": "Stack",
            "Value": "deploy",
          },
          Object {
            "Key": "Stage",
            "Value": Object {
              "Ref": "Stage",
            },
          },
        ],
        "VpcId": Object {
          "Ref": "VpcId",
        },
      },
      "Type": "AWS::EC2::SecurityGroup",
    },
    "RestrictedIngressSecurityGroupAmigotoAMIgoGuHttpsEgressSecurityGroupAmigo34A5A8389000A797C6B3": Object {
      "Properties": Object {
        "Description": "Load balancer to target",
        "DestinationSecurityGroupId": Object {
          "Fn::GetAtt": Array [
            "GuHttpsEgressSecurityGroupAmigo28861996",
            "GroupId",
          ],
        },
        "FromPort": 9000,
        "GroupId": Object {
          "Fn::GetAtt": Array [
            "RestrictedIngressSecurityGroupAmigoF34D371D",
            "GroupId",
          ],
        },
        "IpProtocol": "tcp",
        "ToPort": 9000,
      },
      "Type": "AWS::EC2::SecurityGroupEgress",
    },
    "SSMRunCommandPolicy244E1613": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "ec2messages:AcknowledgeMessage",
                "ec2messages:DeleteMessage",
                "ec2messages:FailMessage",
                "ec2messages:GetEndpoint",
                "ec2messages:GetMessages",
                "ec2messages:SendReply",
                "ssm:UpdateInstanceInformation",
                "ssm:ListInstanceAssociations",
                "ssm:DescribeInstanceProperties",
                "ssm:DescribeDocumentParameters",
                "ssmmessages:CreateControlChannel",
                "ssmmessages:CreateDataChannel",
                "ssmmessages:OpenControlChannel",
                "ssmmessages:OpenDataChannel",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
            Object {
              "Action": Array [
                "ssm:StartSession",
                "ssm:TerminateSession",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "ssm-run-command-policy",
        "Roles": Array [
          Object {
            "Ref": "InstanceRoleAmigo75944747",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "TargetGroupAmigoB9501F07": Object {
      "Properties": Object {
        "HealthCheckIntervalSeconds": 10,
        "HealthCheckPath": "/healthcheck",
        "HealthCheckProtocol": "HTTP",
        "HealthCheckTimeoutSeconds": 5,
        "HealthyThresholdCount": 5,
        "Port": 9000,
        "Protocol": "HTTP",
        "Tags": Array [
          Object {
            "Key": "App",
            "Value": "amigo",
          },
          Object {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          Object {
            "Key": "gu:repo",
            "Value": "guardian/amigo",
          },
          Object {
            "Key": "Stack",
            "Value": "deploy",
          },
          Object {
            "Key": "Stage",
            "Value": Object {
              "Ref": "Stage",
            },
          },
        ],
        "TargetGroupAttributes": Array [
          Object {
            "Key": "deregistration_delay.timeout_seconds",
            "Value": "30",
          },
          Object {
            "Key": "stickiness.enabled",
            "Value": "false",
          },
        ],
        "TargetType": "instance",
        "UnhealthyThresholdCount": 2,
        "VpcId": Object {
          "Ref": "VpcId",
        },
      },
      "Type": "AWS::ElasticLoadBalancingV2::TargetGroup",
    },
    "WazuhSecurityGroup": Object {
      "Properties": Object {
        "GroupDescription": "Allow outbound traffic from wazuh agent to manager",
        "SecurityGroupEgress": Array [
          Object {
            "CidrIp": "0.0.0.0/0",
            "Description": "Wazuh event logging",
            "FromPort": 1514,
            "IpProtocol": "tcp",
            "ToPort": 1514,
          },
          Object {
            "CidrIp": "0.0.0.0/0",
            "Description": "Wazuh agent registration",
            "FromPort": 1515,
            "IpProtocol": "tcp",
            "ToPort": 1515,
          },
        ],
        "Tags": Array [
          Object {
            "Key": "App",
            "Value": "amigo",
          },
          Object {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          Object {
            "Key": "gu:repo",
            "Value": "guardian/amigo",
          },
          Object {
            "Key": "Stack",
            "Value": "deploy",
          },
          Object {
            "Key": "Stage",
            "Value": Object {
              "Ref": "Stage",
            },
          },
        ],
        "VpcId": Object {
          "Ref": "VpcId",
        },
      },
      "Type": "AWS::EC2::SecurityGroup",
    },
  },
}
`;
